Source/Classes/PINRemoteImageDownloadQueue.h

```diff
@@ -0,0 +1,37 @@
+//
+//  PINRemoteImageDownloadQueue.h
+//  PINRemoteImage
+//
+//  Created by Garrett Moon on 3/1/17.
+//  Copyright © 2017 Pinterest. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "PINRemoteImageManager.h"
+
+@class PINURLSessionManager;
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^PINRemoteImageDownloadCompletion)(NSURLResponse *response, NSError *error);
+
+@interface PINRemoteImageDownloadQueue : NSObject
+
+@property (nonatomic, assign) NSUInteger maximumNumberOfOperations;
+
+- (instancetype)init NS_UNAVAILABLE;
+- (PINRemoteImageDownloadQueue *)initWithMaximumNumberOfOperations:(NSUInteger)maximumNumberOfOperations NS_DESIGNATED_INITIALIZER;
+
+- (NSURLSessionDataTask *)addDownloadWithSessionManager:(PINURLSessionManager *)sessionManager
+                                                request:(NSURLRequest *)request
+                                               priority:(PINRemoteImageManagerPriority)priority
+                                      completionHandler:(PINRemoteImageDownloadCompletion)completionHandler;
+
+- (void)dequeueDownload:(NSURLSessionDataTask *)downloadTask;
```

Comment:
It's a bit different than cancel since this class is intended to *only* manage positioning in the queue itself. I.e. dequeuing doesn't actually cancel an inflight network request, that's left up to the caller. Same with priority: it sets the priority in which a task will get executed but does not pass this through to the networking layer, also left up to the caller. I should probably add documentation :P